cdba-server.c

```diff
@@ -99,6 +103,45 @@ static void msg_fastboot_continue(void)
 	cdba_send(MSG_FASTBOOT_CONTINUE);
 }
 
+#ifndef CDBA_CAMERA_DISABLED
+static void capture_image(struct device *device)
+{
+	uint8_t *jpeg = NULL;
+	size_t size = 0;
+
+	if (!device->video_device)
+	{
+		fprintf(stderr, "video_device not specified!\n");
+		return;
+	}
+
+	int ret = camera_capture_jpeg(&jpeg, &size, device->video_device);
+	if (ret < 0)
+	{
+		fprintf(stderr, "Failed to capture image\n");
+		return;
+	}
+
+	const size_t chunk_size = 1024;
+	size_t remaining_size = size;
+	const uint8_t *ptr = jpeg;
+
+	while (remaining_size > 0) {
+		size_t send_size = remaining_size > chunk_size ? chunk_size : remaining_size;
+
+		cdba_send_buf(MSG_CAPTURE_IMAGE, send_size, ptr);
+
+		ptr += send_size;
+		remaining_size -= send_size;
+	}
+
+	// Empty message to indicate end of image
+	cdba_send_buf(MSG_CAPTURE_IMAGE, 0, NULL);
+
+	free(jpeg);
+}
+#endif // CDBA_CAMERA_DISABLED
```

Comment:
```c
#else
static void capture_image(struct device *device)
{
	fprintf(stderr, "Camera support is not built into the server\n");
}
#endif 
```

It is generally more preferable to have stubs / conditionally selected functions rather than spaghetti of `ifdef`'s iside the function body.